tests/LSP6/decodePermissions.test.ts

```diff
@@ -0,0 +1,172 @@
+            PERMISSIONS.CHANGEOWNER,
+            PERMISSIONS.ADDCONTROLLER,
+            PERMISSIONS.CALL,
+            PERMISSIONS.SETDATA,
+        ];
+
+        expect(
+            decodePermissions(
+                toBeHex(
+                    BigInt(PERMISSIONS.CHANGEOWNER) |
+                        BigInt(PERMISSIONS.ADDCONTROLLER) |
+                        BigInt(PERMISSIONS.CALL) |
+                        BigInt(PERMISSIONS.SETDATA),
+                    32,
+                ),
+                'BytesLike',
+            ),
+        ).to.deep.equal(expectedPermissions);
+    });
+
+    it('should pass, testing return value as `bigint[]`', () => {
+        const expectedPermissions = [
+            BigInt(PERMISSIONS.CHANGEOWNER),
+            BigInt(PERMISSIONS.ADDCONTROLLER),
+            BigInt(PERMISSIONS.CALL),
+            BigInt(PERMISSIONS.SETDATA),
+        ];
+
+        expect(
+            decodePermissions(
+                toBeHex(
+                    BigInt(PERMISSIONS.CHANGEOWNER) |
+                        BigInt(PERMISSIONS.ADDCONTROLLER) |
+                        BigInt(PERMISSIONS.CALL) |
+                        BigInt(PERMISSIONS.SETDATA),
+                    32,
+                ),
+                'bigint',
+            ),
+        ).to.deep.equal(expectedPermissions);
+    });
+
+    it('should pass, testing return value as `boolean[]`', () => {
+        // because the `SETDATA` permissions is the 19th `Bit` in the `BitArray`
+        // it means that `SETDATA` would be situated in the 3rd byte from left to right
+        // and because each bytes has 8 bits, the retuned array would have 32 (8 bits * 3 bytes) elements
+        const expectedPermissions = [
+            true,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            true,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+            false,
+        ];
+
+        expect(
+            decodePermissions(
+                toBeHex(
+                    BigInt(PERMISSIONS.CHANGEOWNER) |
+                        BigInt(PERMISSIONS.ADDCONTROLLER) |
+                        BigInt(PERMISSIONS.CALL) |
+                        BigInt(PERMISSIONS.SETDATA),
+                    32,
+                ),
+                'boolean',
+            ),
+        ).to.deep.equal(expectedPermissions);
+    });
+
+    it('should pass, testing return value as `LSP6PermissionName[]`', () => {
+        const expectedPermissions = ['CHANGEOWNER', 'ADDCONTROLLER', 'CALL', 'SETDATA'];
```

Comment:
Representing the permissions should probably ether be an object or a Set(). I.e. either { 'CHANGEOWNER': true } or new Set(['CHANGEOWNER']) otherwise the result would be hard to use.